hax/treemakers/se_treemaker.py

```diff
@@ -0,0 +1,64 @@
+import hax
+import numpy as np
+from pax import units
+
+
+def yield_single_electrons(event, nhits_bounds, width_bounds):
+    # Get all non-lone-hit peaks in the TPC
+    peaks = [p for p in event.peaks if p.detector == 'tpc' and not p.type == 'lone_hit']
+    peaks = sorted(peaks, key=lambda p: p.left)
+
+    if not len(peaks):
+        return []
+
+    # For each of these, find the smallest gap
+    lefts, rights = np.array([(p.left, p.right) for p in peaks]).T * 10 * units.ns
+    gap_on_left = np.concatenate(([0], lefts[1:] - rights[:-1]))
+    gap_on_right = np.concatenate((lefts[1:] - rights[:-1], [0]))
+    smallest_gap = np.clip(gap_on_left, 0, gap_on_right)
+
+    # ses = []
+    # time_to_nearest = []
+    for i, peak in enumerate(peaks):
+        time_to_nearest_peak = smallest_gap[i]
+        if time_to_nearest_peak < 10 * units.us:
+            continue
+        if not (nhits_bounds[0] <= peak.n_hits < nhits_bounds[1]):
+            continue
+        width = peak.range_area_decile[5]
+        if not (width_bounds[0] <= width < width_bounds[1]):
+            continue
+        yield peak, time_to_nearest_peak
+
+
+class IsolatedPeaks(hax.minitrees.MultipleRowExtractor):
+    __version__ = '0.1.2'
```

Comment:
Please add a docstring explaining what this class does.  Title and few sentences describing it.